src/lib.rs

```diff
@@ -419,8 +496,19 @@ where
     }
 
     fn on_close(&self, id: Id, ctx: Context<S>) {
+        let bufs = &mut *self.bufs.lock().unwrap();
+
+        // Store the most recently entered span
+        let _ = self.current_span.compare_exchange(
```

Comment:
I am not entirely sure what the new value should be here.

Closing a span won't necessarily mean that the parent span is the new active one, as any span could be printed next on a different thread. I think for non-concurrent usage the parent of a span will most often be the currently active span, or it will be closed immediately afterwards (SeqCst).

What are you thoughts @davidbarsky?